vault/external_tests/replication_binary/replication_ent_test.go

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package replication_binary
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/vault/sdk/helper/testcluster"
+	"github.com/hashicorp/vault/sdk/helper/testcluster/docker"
+)
+
+func TestReplication_FailoverPrimaryActive(t *testing.T) {
```

Comment:
No need to review this here, it's moving to an ent PR shortly (though I'll have to merge the remainder of this one first, since it depends on these changes.)